include/System/ParamInst.hpp

```diff
@@ -1,13 +1,35 @@
 #ifndef PARAM_INST_HPP
 #define PARAM_INST_HPP
 
+#include "JSystem/JDrama/JDRNameRef.hpp"
 #include "System/BaseParam.hpp"
 
+#define PARAM_INIT(member, value)                                              \
+	member(this, JDrama::TNameRef::calcKeyCode("#member"), "#member", value)
+
 template <typename T> class TParamT : public TBaseParam {
 public:
+	TParamT(TParams* params, unsigned short code, const char* paramName,
+	        T defaultValue)
+	    : TBaseParam(params, code, paramName)
+	    , value(defaultValue)
+	{
+	}
+
 	void load(JSUMemoryInputStream& stream);
```

Comment:
please mark virtual functions as virtual even when overriding for clarity (I'm against it when writing new code, but for decomp it's quite important I think)

Reply:
I had forgotten your life lesson of gw 😅